src/os/filestore/FileStore.cc

```diff
@@ -1794,11 +1811,249 @@ int FileStore::umount()
     timer.shutdown();
   }
 
+#ifdef HAVE_LIBAIO
+  destroy_read_completion_threads();
+#endif
+
   // nothing
   return 0;
 }
 
+#ifdef HAVE_LIBAIO
+int FileStore::create_read_completion_threads()
+{
+  // NB: m_filestore_async_threads may be 0.
+  int error = 0;
+  dout(0) << "FileStore: creating " << m_filestore_async_threads
+	  << " read completion threads." << dendl;
+  for (int i = 0; i < m_filestore_async_threads; ++i) {
+    ReadCompletionThread *thr = new ReadCompletionThread(this);
+    /*
+     * FIXME: do we need to set priority?
+     * int r = thr->set_ioprio(thr->ioprio_class, thr->ioprio_priority);
+     */
+    read_completion_threads.push_back(thr);
+    error = thr->Start();
+    if (error)
+      break;
+  }
+
+  if (error) {
+    derr << "unable to create read completion threads, error = "
+	 << error << dendl;
+    for (auto rct : read_completion_threads)
+      delete rct;
```

Comment:
If you change to vector of unique_ptr, no need to manually delete here, the clear() will suffice.